x-pack/plugins/ml/public/data_frame/components/define_pivot/define_pivot_form.tsx

```diff
@@ -0,0 +1,248 @@
+  const pivotQuery = getPivotQuery(search);
+
+  const valid = pivotGroupBy.length > 0 && aggList.length > 0;
+
+  useEffect(
+    () => {
+      onChange({ aggList, aggs: pivotAggs, groupBy: pivotGroupBy, search, valid });
+    },
+    [
+      aggList,
+      pivotAggs.map(d => `${d.agg} ${d.field} ${d.formRowLabel}`).join(' '),
+      pivotGroupBy,
+      search,
+      valid,
+    ]
+  );
+
+  const displaySearch = search === defaultSearch ? emptySearch : search;
+
+  return (
+    <EuiFlexGroup>
+      <EuiFlexItem grow={false} style={{ minWidth: '420px' }}>
+        <EuiForm>
+          <EuiFormRow
+            label={i18n.translate('xpack.ml.dataframe.definePivotForm.queryLabel', {
+              defaultMessage: 'Query',
+            })}
+          >
+            <EuiFieldSearch
+              placeholder={i18n.translate('xpack.ml.dataframe.definePivotForm.queryPlaceholder', {
+                defaultMessage: 'Search...',
+              })}
+              onChange={searchHandler}
+              value={displaySearch}
+            />
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={i18n.translate('xpack.ml.dataframe.definePivotForm.groupByLabel', {
+              defaultMessage: 'Group by',
+            })}
+          >
+            <Fragment>
+              <GroupByList list={pivotGroupBy} deleteHandler={deleteGroupBy} />
+              <DropDown
+                changeHandler={addGroupBy}
+                options={groupByOptions}
+                placeholder={i18n.translate(
+                  'xpack.ml.dataframe.definePivotForm.groupByPlaceholder',
+                  {
+                    defaultMessage: 'Add a group by field ...',
+                  }
+                )}
+              />
+            </Fragment>
+          </EuiFormRow>
+
+          <EuiFormRow
+            label={i18n.translate('xpack.ml.dataframe.definePivotForm.aggregationsLabel', {
+              defaultMessage: 'Aggregations',
+            })}
+          >
+            <Fragment>
+              <AggListForm
+                list={aggList}
+                optionsData={aggOptionsData}
+                deleteHandler={deleteAggregation}
+              />
+              <DropDown
+                changeHandler={addAggregation}
+                options={aggOptions}
+                placeholder={i18n.translate(
+                  'xpack.ml.dataframe.definePivotForm.aggregationsPlaceholder',
+                  {
+                    defaultMessage: 'Add an aggregation ...',
+                  }
+                )}
+              />
+            </Fragment>
+          </EuiFormRow>
+          {!valid && (
+            <EuiFormHelpText style={{ maxWidth: '320px' }}>
+              {i18n.translate('xpack.ml.dataframe.definePivotForm.formHelp', {
+                defaultMessage:
+                  'Data Frame Transforms are scalable and automated processes for pivoting. Choose at least one group-by and aggregation to get started.',
```

Comment:
I think this should be sentence case `Data frame transforms...`

Reply:
Done in 6b4b1f4.